### PR TITLE
Hotfix/file destination write

### PR DESCRIPTION
**Pull request by jayckaiser**
# Hotfix: File Destination Write Fix

## Description & motivation
This PR fixes a bug where rows written on `FileDestination.execute()` were malformed due to parallel writes occurring during `dask.DataFrame.apply(IOTextWrapper.write)`.

## PR Merge Priority:

- [ ] Low
- [ ] Medium
- [x] High

This is a production-level break that affects the public PIP version as well.

## Changes to existing files:
- `earthmover/nodes/destination.py`
  - Compute and bulk-write dataframe partitions, instead of iterating and writing each row.
  - Clean up `render_row()` logic to improve at-sight visibility.

## Tests and QC done:

Tests reran successfully. Original Earthmover assessment bundle from which this bug was found was resolved.

### CODE REVIEW

File: earthmover/nodes/destination.py [hunk 1]
```diff
@@ -104,7 +104,9 @@ def execute(self, **kwargs):
             if self.header:
                 fp.write(self.header)
 
-            self.data.apply(lambda row: fp.write(row + '\n'), meta=pd.Series('string')).compute()
+            for partition in self.data.partitions:
+                fp.writelines(partition.compute())
+                partition = None  # Remove partition from memory immediately after write.
```
Comment (johncmerfeld) [inline earthmover/nodes/destination.py:109]: Why is this necessary?

**Review by johncmerfeld — APPROVED** [review]
Looks good to me!

File: earthmover/nodes/destination.py (hunk 1 above)
Comment (jayckaiser) [reply earthmover/nodes/destination.py:109]: That's right. If we don't do this, `partition` will linger in memory at least until the end of the execute. I don't want any residual objects to be saved.
